server/hooks.go

```diff
@@ -62,10 +62,17 @@ func (p *Plugin) MessageHasBeenPosted(c *plugin.Context, post *model.Post) {
 		// Still appear to the end user as if their feedback was actually sent
 	}
 
-	// Respond to the feedback
+	rootID := post.RootId
+	// if it is a new post in the channel, update response RootId
+	if post.RootId == "" {
+		rootID = post.Id
+	}
```

Comment:
As far as I understand the ticket, this is not the correct logic. I think it should be
- If the user did reply to the existing post, also rely to this post in a thread.
- If the user didn't rely to the existing post i.e. posted a new message in the main panel, post the response also in the main panel.

@esadur Can you confirm that my understanding of the ticket is correct?

Reply:
@hanzei not quite, based on the ticket spec: "the responses as a reply to the users feedback". So this means

1. If the user did reply to the existing post, also rely to this post in the thread.
2. If the user didn't rely to the existing post i.e. posted a new message in the main panel, post the surveybot response **as a reply to the users message**

Reply:
This logic is correct then. If the user posted in a thread (`post.RootId` is not empty), then reply to the same thread (set `rootID` to `post.RootId`), otherwise, reply to the user's post (set `rootID` to `post.Id`).